azurerm/internal/services/datashare/resource_arm_data_share.go

```diff
@@ -0,0 +1,306 @@
+	}
+
+	d.SetId(*resp.ID)
+
+	if d.HasChange("snapshot_schedule") {
+		// only one dependent sync setting is allowed in one data share
+		o, _ := d.GetChange("snapshot_schedule")
+		if origins := o.([]interface{}); len(origins) > 0 {
+			syncFuture, err := syncClient.Delete(ctx, accountId.ResourceGroup, accountId.Name, name, origins[0].(map[string]interface{})["name"].(string))
+			if err != nil {
+				return fmt.Errorf("deleting DataShare %q snapshot schedule (Resource Group %q / accountName %q): %+v", name, accountId.ResourceGroup, accountId.Name, err)
+			}
+			if err = syncFuture.WaitForCompletionRef(ctx, syncClient.Client); err != nil {
+				return fmt.Errorf("waiting for DataShare %q snapshot schedule (Resource Group %q / accountName %q) to be deleted: %+v", name, accountId.ResourceGroup, accountId.Name, err)
+			}
+		}
+	}
+
+	if snapshotSchedule := expandAzureRmDataShareSnapshotSchedule(d.Get("snapshot_schedule").([]interface{})); snapshotSchedule != nil {
+		if _, err := syncClient.Create(ctx, accountId.ResourceGroup, accountId.Name, name, d.Get("snapshot_schedule.0.name").(string), snapshotSchedule); err != nil {
+			return fmt.Errorf("creating DataShare %q snapshot schedule (Resource Group %q / accountName %q): %+v", name, accountId.ResourceGroup, accountId.Name, err)
+		}
+	}
+	return resourceArmDataShareRead(d, meta)
+}
+
+func resourceArmDataShareRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataShare.SharesClient
+	accountClient := meta.(*clients.Client).DataShare.AccountClient
+	syncClient := meta.(*clients.Client).DataShare.SynchronizationClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DataShareID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.AccountName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] DataShare %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving DataShare %q (Resource Group %q / accountName %q): %+v", id.Name, id.ResourceGroup, id.AccountName, err)
+	}
+
+	accountResp, err := accountClient.Get(ctx, id.ResourceGroup, id.AccountName)
+	if err != nil {
+		return fmt.Errorf("retrieving DataShare Account %q (Resource Group %q): %+v", id.AccountName, id.ResourceGroup, err)
+	}
+	if accountResp.ID == nil || *accountResp.ID == "" {
+		return fmt.Errorf("reading DataShare Account %q (Resource Group %q): ID is empty", id.AccountName, id.ResourceGroup)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("account_id", accountResp.ID)
+
+	if props := resp.ShareProperties; props != nil {
+		d.Set("share_kind", props.ShareKind)
+		d.Set("description", props.Description)
+		d.Set("terms", props.Terms)
+	}
+
+	if syncIterator, err := syncClient.ListByShareComplete(ctx, id.ResourceGroup, id.AccountName, id.Name, ""); syncIterator.NotDone() {
```

Comment:
Should be not be using the name property to look this up with a `.Get` instead of listing them all like this?

Reply:
If so, we need to ignore all the fields in the block in acctest `data.ImportStep()`, which means that the block will be empty if we `terraform import` the resource

Reply:
That's a great point! didn't take that into consideration